### PR TITLE
chore: refactor Windows image build and add test

### PR DESCRIPTION
**Pull request by andyzhangx**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
We don't have any test coverage with Windows image build, this PR refactors Windows image build and add Windows image build in github action

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
none
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -69,6 +62,12 @@ build-container-linux-%:
 		--build-arg OS=linux \
 		--build-arg ARCH=$* .
 
+build-container-windows-%:
```
Comment (mauriciopoppe) [inline Makefile:65]: I remember that @mattcary had problems building a Windows image without pushing it to a registry, did you have any issues running this command?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -37,5 +37,5 @@ Fixes #
 
 **Release note**:
 ```release-note
-
+none
```
Comment (mauriciopoppe) [inline .github/PULL_REQUEST_TEMPLATE.md:40]: I think this shouldn't be the default value

File: Makefile [hunk 3]
```diff
@@ -81,7 +80,7 @@ build-and-push-container-windows-%: init-buildx
 	CGO_ENABLED=0 GOOS=windows GOARCH=amd64 go build -a -ldflags='-extldflags="-static" -X="main.version=${STAGINGVERSION}"' -mod vendor -o _output/windows/amd64/local-volume-provisioner.exe ./cmd/local-volume-provisioner
 	$(DOCKER) buildx build --file=./deployment/docker/Dockerfile.Windows --platform=windows/amd64 \
 		-t $(STAGINGIMAGE):$(STAGINGVERSION)_windows_$* \
-		--build-arg BASE_IMAGE=$(call lookup,$*,$(WINDOWS_DISTROS),$(WINDOWS_BASE_IMAGES)) \
+		--build-arg OSVERSION=$* \
```
Comment (mauriciopoppe) [inline Makefile:83]: OSVERSION isn't set anywhere where this script runs, please take a look at https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/blob/master/hack/release.sh#L205

Comment (andyzhangx) [reply Makefile:83]: thanks, let me change `release.sh` script too, `BASE_IMAGE` depends on `lookup` func, I think that's difficult to maintain.

**Review by ZeroMagic — APPROVED** [review]
/lgtm

File: Makefile (hunk 1 above)
Comment (andyzhangx) [reply Makefile:65]: Hi @mauriciopoppe 
I could remove this make `build-container-windows-%` command in Makefile, actually it does not change the default behavior, the Windows Dockerfile build works well now which shows the `OSVersion` refactor change works well:
https://k8s-testgrid.appspot.com/sig-storage-image-build#post-sig-storage-local-static-provisioner-push-images

Comment (mauriciopoppe) [reply Makefile:65]: great, thanks for the clarification

**Pull request merged**